### PR TITLE
Rename UpdateMetadta with PrepareMetadata and add comments

### PR DESCRIPTION
**Pull request by qinezh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Plugins/IPostProcessor.cs [hunk 1]
```diff
@@ -7,8 +7,14 @@ namespace Microsoft.DocAsCode.Plugins
...
+        /// <summary>
+        /// Update global metadata before building all the files declared in `docfx.json`
+        /// </summary>
+        ImmutableDictionary<string, object> PrepareMetadata(ImmutableDictionary<string, object> metadata);
 
-        Manifest Process(Manifest manifest, string baseDir);
+        /// <summary>
+        /// Add/remove/update all the files included in manifest
+        /// </summary>
+        Manifest Process(Manifest manifest, string outputFolder);
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Plugins/IPostProcessor.cs]: >outputFolder [](start = 51, length = 12)

also add description for parameters?

**Comment by hellosnow** [comment]
:shipit:

**Comment by vwxyzh** [comment]
👍

**Comment by superyyrrzz** [comment]
:panda_face:

**Pull request merged**